lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -1538,7 +1538,17 @@ namespace Js
             // else use enumerator to extract keys from source
             else
             {
-                AssignForGenericObjects(from, to, scriptContext);
+                DynamicObject* fromObj = JavascriptOperators::TryFromVar<DynamicObject>(from);
+                DynamicObject* toObj = JavascriptOperators::TryFromVar<DynamicObject>(to);
+                bool cloned = false;
```

Comment:
say we failed to clone for the first source, do we still attempt to do for the next sources?

Reply:
Do we need to check for cross-site object and just don't try that?

---
In reply to: [174883683](https://github.com/Microsoft/ChakraCore/pull/4817#discussion_r174883683) [](ancestors = 174883683)

Reply:
Should we exclude external objects (IsExternal()). Not sure what is the harm though.

---
In reply to: [174883937](https://github.com/Microsoft/ChakraCore/pull/4817#discussion_r174883937) [](ancestors = 174883937,174883683)

Reply:
I'll add case for cross-site object, I'm not sure about external, but I'll exclude them for good measure